tests/integration/api/test_api.py

```diff
@@ -521,221 +357,156 @@ def _compare_screenboard(apiBoard, expectedBoard):
-        public_url = share_res['public_url']
+        share_res = dog.Screenboard.share(get_res["id"])
+        assert share_res["board_id"] == get_res["id"]
+        public_url = share_res["public_url"]
 
         response = requests.get(public_url)
         assert response.status_code == 200
 
-        dog.Screenboard.revoke(get_res['id'])
+        dog.Screenboard.revoke(get_res["id"])
         response = requests.get(public_url)
         assert response.status_code == 404
 
-        delete_res = dog.Screenboard.delete(update_res['id'])
-        assert delete_res['id'] == update_res['id']
+        delete_res = dog.Screenboard.delete(update_res["id"])
+        assert delete_res["id"] == update_res["id"]
 
-    @pytest.mark.monitor
     def test_monitor_crud(self):
         # Metric alerts
         query = "avg(last_1h):sum:system.net.bytes_rcvd{host:host0} > 100"
 
         options = {
-            'silenced': {'*': int(time.time()) + 60 * 60},
-            'notify_no_data': False
+            "silenced": {"*": int(time.time()) + 60 * 60},
+            "notify_no_data": False
         }
-        monitor_id = dog.Monitor.create(type='metric alert', query=query, options=options)['id']
-        monitor = dog.Monitor.get(monitor_id)
+        monitor = dog.Monitor.create(type="metric alert", query=query, options=options)
 
-        eq(monitor['query'], query)
-        eq(monitor['options']['notify_no_data'],
-            options['notify_no_data'])
-        eq(monitor['options']['silenced'], options['silenced'])
+        assert monitor["query"] == query
+        assert monitor["options"]["notify_no_data"] == options["notify_no_data"]
+        assert monitor["options"]["silenced"] == options["silenced"]
 
         query2 = "avg(last_1h):sum:system.net.bytes_rcvd{host:host0} > 200"
-        updated_monitor_id = dog.Monitor.update(monitor_id, query=query2, options=options)['id']
-        monitor = dog.Monitor.get(updated_monitor_id)
-        eq(monitor['query'], query2)
+        monitor = dog.Monitor.update(monitor["id"], query=query2, options=options)
+        assert monitor["query"] == query2
+        assert monitor["options"]["notify_no_data"] == options["notify_no_data"]
+        assert monitor["options"]["silenced"] == options["silenced"]
+
+        name = "test_monitors"
+        monitor = dog.Monitor.update(monitor["id"], query=query2, name=name, options={"notify_no_data": True})
+        assert monitor["name"] == name
+        assert monitor["query"] == query2
+        assert monitor["options"]["notify_no_data"] is True
 
-        name = 'test_monitors'
-        monitor_id = dog.Monitor.update(monitor_id, query=query2, name=name,
-                                        options={'notify_no_data': True})['id']
-        monitor = dog.Monitor.get(monitor_id)
-        eq(monitor['name'], name)
-        eq(monitor['options']['notify_no_data'], True)
+        monitors = [m for m in dog.Monitor.get_all() if m["id"] == monitor["id"]]
```

Comment:
What is this testing? IDs should be unique so would this ever not be `1`? Is this more testing that the `get_all` function actually returns a monitor object?

Reply:
yup, testing the `get_all`